spec/std/yaml/from_yaml_spec.cr

```diff
@@ -0,0 +1,397 @@
+require "spec"
+require "yaml"
+
+private class ObjectStub
+  def self.new(pp : YAML::PullParser)
+    pp.read_scalar.should eq "hello"
+    new
+  end
+end
+
+describe "Object.from_yaml" do
+  it "should create call .new on with a YAML::PullParser" do
+    ObjectStub.from_yaml("hello")
+  end
+end
+
+describe "Nil.from_yaml" do
+  values = ["", "NULL", "Null", "null", "~"]
+  values.each do |value|
+    it "should serialize and deserialize" do
+      Nil.from_yaml(nil.to_yaml).should eq nil
+    end
+
+    it "should return nil if #{value}" do
+      # Test will an array since a standalone empty value shows as STREAM_END
+      Array(Nil).from_yaml("- " + value).should eq [nil]
+    end
+
+    it "should raise if #{value} is quoted with double quotes" do
+      expect_raises(YAML::ParseException) do
+        Nil.from_yaml "\"#{value}\""
+      end
+    end
+
+    it "should raise if #{value} is quoted with single quotes" do
+      expect_raises(YAML::ParseException) do
+        Nil.from_yaml "'#{value}'"
+      end
+    end
+  end
+
+  it "should raise if not a null value" do
+    expect_raises(YAML::ParseException) do
+      Nil.from_yaml "hello"
+    end
+  end
+end
+
+describe "Bool.from_yaml" do
+
+  describe "true values" do
+    it "should serialize and deserialize" do
+      Bool.from_yaml(true.to_yaml).should eq true
+    end
+
+    values = ["true", "True", "TRUE", "on", "On", "ON", "y", "Y", "yes", "Yes", "YES"]
```

Comment:
The only downside to this is that if there is a particular failure on one of the values, but not the other values, I don't think you'll be able to differentiate that in the error message.  That is you'll see something like this:

```text
  1) Bool.from_yaml true values should return true if a truthy value
     Failure/Error: Bool.from_yaml(value).should eq true

       expected: true
            got: false

     # ./spec/std/yaml/from_yaml_spec.cr:69

Finished in 0.69 milliseconds
1 examples, 1 failures, 0 errors, 0 pending
```

When I've done this in other projects, I've always been forced to append some custom message to the expectation that says "for value #{foo}", so I can tell them apart.

Additionally, if the first value fails, then it never tests the other values, essentially making it fail-fast.

@straight-shoota Thoughts in general to this?  Maybe it doesn't apply in this particular case, but I'm curious what you think overall.